Smt/Reconstruction/Rewrites/Prop.lean

```diff
@@ -7,49 +7,107 @@ Authors: Abdalrhman Mohamed
 
 import Smt.Reconstruction.Rewrites.Simp
 
+private theorem And.assoc : (p ∧ q) ∧ r ↔ p ∧ (q ∧ r) := .intro
+  (let ⟨⟨hp, hq⟩, hr⟩ := ·; .intro hp (.intro hq hr))
+  (let ⟨hp, ⟨hq, hr⟩⟩ := ·; .intro (.intro hp hq) hr)
+
+private theorem Or.comm : p ∨ q ↔ q ∨ p :=
+  .intro (·.elim Or.inr Or.inl) (·.elim Or.inr Or.inl)
+
+private theorem Or.assoc : (p ∨ q) ∨ r ↔ p ∨ (q ∨ r) := .intro
+  (·.elim (·.elim .inl (.inr ∘ .inl)) (.inr ∘ .inr))
+  (·.elim (.inl ∘ .inl) (·.elim (.inl ∘ .inr) .inr))
+
```

Comment:
Why not use Std.Logic.and_assoc, or_comm and or_assoc?

Reply:
I did so because I prefer to avoid unnecessary imports. Not a strong reason, so I am switching to Std theorems.